### ISSUE TITLE
Potential efficiency problem in HMM_Normalize

### ISSUE THREAD

**Issue by bvisness** [StrangeZak/Handmade-Math#32]
Here is the full function body for `HMM_Normalize(hmm_vec3)`:

```c++
HINLINE hmm_vec3
HMM_Normalize(hmm_vec3 A)
{
    hmm_vec3 Result = {0};

    Result.X = A.X * (1.0f / HMM_Length(A));
    Result.Y = A.Y * (1.0f / HMM_Length(A));
    Result.Z = A.Z * (1.0f / HMM_Length(A));
    
    return (Result);
}
```

`HMM_Length(A)` is called three times during this process, even though the value is the same in all three cases.

I am no expert on what low-level shenanigans are in play when this code is compiled, but I suspect that compilers would not be able to optimize this. We should save the result and then use it in all three divisions.

**Comment by StrangeZak** [comment]
I mean its probably not a big deal, but its an easy fix, so it should be done.

**Comment by bvisness** [comment]
Fixed in #37.
